net/dns/manager.go

```diff
@@ -207,7 +207,7 @@ func (m *Manager) compileConfig(cfg Config) (rcfg resolver.Config, ocfg OSConfig
 		// case where cfg is entirely zero, in which case these
 		// configs clear all Tailscale DNS settings.
 		return rcfg, ocfg, nil
-	case cfg.hasDefaultIPResolversOnly():
+	case cfg.hasDefaultIPResolversOnly() && !cfg.hasHostsWithoutSplitDNSRoutes():
```

Comment:
comment here describing why?

Reply:
I dunno, the next case's comment basically already says it:

```
	// Default resolvers plus other stuff always ends up proxying
	// through quad-100.
```

"plus other stuff"